Yatzy.Tests/RoundTests.cs

```diff
@@ -0,0 +1,119 @@
+        _turnFactoryMock.Setup(turn => turn.CreateTurn(_player2)).Returns(_player2TurnMock.Object);
+        _player1TurnMock.Setup(playerTurn => playerTurn.PlayerTurn()).Verifiable();
+        _player2TurnMock.Setup(playerTurn => playerTurn.PlayerTurn()).Verifiable();
+        
+
+        //Act
+        _round.PlayTurns();
+
+        //Assert
+        _turnFactoryMock.Verify();
+    }
+    
+    public static IEnumerable<object[]> RoundWinnerTestObject()
+    {
+        yield return new object[] { 50, new List<int> {5,5,5,5,5}, new List<int> {1,2,3,3,3} };
+        yield return new object[] { 40, new List<int> {1,2,3,4,5}, new List<int> {3,3,3,2,1} };
+        yield return new object[] { 30, new List<int> {1,2,3,4,1}, new List<int> {1,3,3,1,5} };
+        yield return new object[] { 21, new List<int> {4,4,4,4,5}, new List<int> {1,2,3,2,6} };
+    }
+    
+    [Theory]
+    [MemberData(nameof(RoundWinnerTestObject))]
+    public void GivenARound_WhenGetRoundResultIsCalledWhereThereIsAPlayerThatScoredMorePointsThanOtherPlayers_ThenReturnTheRoundWinner(int score, List<int> diceHandOne, List<int> diceHandTwo)
+    {
+
+        //Arrange
+        _turnFactoryMock.Setup(turn => turn.CreateTurn(_player1)).Returns(_player1TurnMock.Object);
+        _turnFactoryMock.Setup(turn => turn.CreateTurn(_player2)).Returns(_player2TurnMock.Object);
+        _mockPlayer1RandomDieRoll.Setup(diceHand => diceHand.GetDiceHand()).Returns(diceHandOne);
+        _mockPlayer2RandomDieRoll.Setup(diceHand => diceHand.GetDiceHand()).Returns(diceHandTwo);
+        var dummyInt = 10;
+        _player1.CompleteTurn();
+        _player2.CompleteTurn();
+        _player1.UpdateRecordsAfterTurn(dummyInt);
+        _player2.UpdateRecordsAfterTurn(dummyInt);
+        var expectedWinner = new RoundWinner(score, _player1);
+        
+        //Act
+        _round.PlayTurns();
+        var roundResult = _round.GetRoundResult();
+
+        //Assert
+        roundResult.Should().BeEquivalentTo(expectedWinner);
+    }
+    
+    public static IEnumerable<object[]> RoundTieTestObject()
+    {
+        yield return new object[] { 50, new List<int> {5,5,5,5,5}, new List<int> {5,5,5,5,5} };
+        yield return new object[] { 40, new List<int> {1,2,3,4,5}, new List<int> {1,2,3,4,5} };
+        yield return new object[] { 30, new List<int> {1,2,3,4,1}, new List<int> {1,2,3,4,1} };
+        yield return new object[] { 21, new List<int> {4,4,4,4,5}, new List<int> {4,4,4,4,5} };
+    }
+    
+    [Theory]
+    [MemberData(nameof(RoundTieTestObject))]
+    public void GivenARound_WhenGetRoundResultIsCalledWhereTheHighestScoreIsAccompaniedByTwoOrMorePlayers_ThenReturnARoundTie(int score, List<int> diceHandOne, List<int> diceHandTwo)
+    {
+
+        //Arrange
+        _turnFactoryMock.Setup(turn => turn.CreateTurn(_player1)).Returns(_player1TurnMock.Object);
+        _turnFactoryMock.Setup(turn => turn.CreateTurn(_player2)).Returns(_player2TurnMock.Object);
```

Comment:
Overall this is quite good an readable 👍 if you're curious you could potentially look into something like the builder pattern to reduce duplication and make the code slightly more readable. no need to implement it but it's good to be aware of it